File: student-management-dashboard/services/permission.service.ts
import { api } from './api'
import type { Permission } from '~/types'

export const permissionService = {
  async getAll() {
    const response = await api.get<any>('/permissions')
    return response.content || response.data || []
  },

  async getById(id: string) {
    return await api.get<Permission>(`/permissions/${id}`)
  },

  async create(data: Partial<Permission>) {
    return await api.post<Permission>('/permissions', data)
  },

  async update(id: string, data: Partial<Permission>) {
    return await api.put<Permission>(`/permissions/${id}`, data)
  },

  async delete(id: string) {
    return await api.delete(`/permissions/${id}`)
  }
}

File: student-management-api/src/main/java/com/kh/rupp_dev/studentmanagement/service/PermissionService.java
package com.kh.rupp_dev.studentmanagement.service;

import com.kh.rupp_dev.studentmanagement.dto.request.PermissionRequest;
import com.kh.rupp_dev.studentmanagement.dto.response.PermissionResponse;

import java.util.List;

public interface PermissionService {

    PermissionResponse create(PermissionRequest request);

    PermissionResponse update(Integer id , PermissionRequest request);

    void delete(Integer id);

    List<PermissionResponse> getAll();

    PermissionResponse getById(Integer id);

    List<PermissionResponse> findByModule(String module);
}